opensrp-opd/src/main/java/org/smartregister/opd/repository/OpdTreatmentDetailRepository.java

```diff
@@ -0,0 +1,79 @@
+            + OpdDbConstants.Column.OpdTreatmentDetail.SPECIAL_INSTRUCTIONS + " VARCHAR NULL, "
+            + OpdDbConstants.Column.OpdTreatmentDetail.VISIT_ID + " VARCHAR NOT NULL, "
+            + OpdDbConstants.Column.OpdTreatmentDetail.CREATED_AT + " VARCHAR NULL )";
+
+    private static final String INDEX_BASE_ENTITY_ID = "CREATE INDEX " + OpdDbConstants.Table.OPD_TREATMENT_DETAIL
+            + "_" + OpdDbConstants.Column.OpdTreatmentDetail.BASE_ENTITY_ID + "_index ON " + OpdDbConstants.Table.OPD_TREATMENT_DETAIL +
+            "(" + OpdDbConstants.Column.OpdTreatmentDetail.BASE_ENTITY_ID + " COLLATE NOCASE);";
+
+    private static final String INDEX_VISIT_ID = "CREATE INDEX " + OpdDbConstants.Table.OPD_TREATMENT_DETAIL
+            + "_" + OpdDbConstants.Column.OpdTreatmentDetail.VISIT_ID + "_index ON " + OpdDbConstants.Table.OPD_TREATMENT_DETAIL +
+            "(" + OpdDbConstants.Column.OpdTreatmentDetail.VISIT_ID + " COLLATE NOCASE);";
+
+    public static void createTable(@NonNull SQLiteDatabase database) {
+        database.execSQL(CREATE_TABLE_SQL);
+        database.execSQL(INDEX_BASE_ENTITY_ID);
+        database.execSQL(INDEX_VISIT_ID);
+    }
+
+    @Override
+    public boolean saveOrUpdate(@NonNull org.smartregister.opd.pojo.OpdTreatmentDetail opdTreatmentDetail) {
+        ContentValues contentValues = new ContentValues();
+        contentValues.put(OpdDbConstants.Column.OpdTreatmentDetail.BASE_ENTITY_ID, opdTreatmentDetail.getBaseEntityId());
+        contentValues.put(OpdDbConstants.Column.OpdTreatmentDetail.MEDICINE, opdTreatmentDetail.getMedicine());
+        contentValues.put(OpdDbConstants.Column.OpdTreatmentDetail.DOSAGE, opdTreatmentDetail.getDosage());
+        contentValues.put(OpdDbConstants.Column.OpdTreatmentDetail.DURATION, opdTreatmentDetail.getDuration());
+        contentValues.put(OpdDbConstants.Column.OpdTreatmentDetail.FREQUENCY, opdTreatmentDetail.getFrequency());
+        contentValues.put(OpdDbConstants.Column.OpdTreatmentDetail.NOTE, opdTreatmentDetail.getNote());
+        contentValues.put(OpdDbConstants.Column.OpdTreatmentDetail.VISIT_ID, opdTreatmentDetail.getVisitId());
+        contentValues.put(OpdDbConstants.Column.OpdTreatmentDetail.PROPERTY, opdTreatmentDetail.getProperty());
+        contentValues.put(OpdDbConstants.Column.OpdTreatmentDetail.SPECIAL_INSTRUCTIONS, opdTreatmentDetail.getSpecialInstructions());
+        contentValues.put(OpdDbConstants.Column.OpdTreatmentDetail.TREATMENT_TYPE, opdTreatmentDetail.getTreatmentType());
+        contentValues.put(OpdDbConstants.Column.OpdTreatmentDetail.TREATMENT_TYPE_SPECIFY, opdTreatmentDetail.getTreatmentTypeSpecify());
+        contentValues.put(OpdDbConstants.Column.OpdTreatmentDetail.CREATED_AT, opdTreatmentDetail.getCreatedAt());
+        SQLiteDatabase sqLiteDatabase = getWritableDatabase();
+        long rows = sqLiteDatabase.insert(OpdDbConstants.Table.OPD_TREATMENT_DETAIL, null, contentValues);
```

Comment:
I also think this call to the DB should `addOrUpdate`. I am not sure if `insert` does this

Reply:
No it doesn't but am updating the function to only say save, so that there can be no confusion.